javascript/packages/fury/lib/reader/index.ts

```diff
@@ -22,175 +22,184 @@ import { isNodeEnv } from "../util";
 
-  function float32() {
-    const result = dataView.getFloat32(cursor, true);
-    cursor += 4;
+  float32() {
+    const result = this.dataView.getFloat32(this.cursor, true);
+    this.cursor += 4;
     return result;
   }
 
-  function float64() {
-    const result = dataView.getFloat64(cursor, true);
-    cursor += 8;
+  float64() {
+    const result = this.dataView.getFloat64(this.cursor, true);
+    this.cursor += 8;
     return result;
   }
 
-  function stringUtf8At(start: number, len: number) {
-    return buffer.toString("utf8", start, start + len);
+  stringUtf8At(start: number, len: number) {
+    return this.platformBuffer.toString("utf8", start, start + len);
   }
 
-  function stringUtf8(len: number) {
-    const result = buffer.toString("utf8", cursor, cursor + len);
-    cursor += len;
+  stringUtf8(len: number) {
+    const result = this.platformBuffer.toString("utf8", this.cursor, this.cursor + len);
+    this.cursor += len;
     return result;
   }
 
-  function stringOfVarUInt32() {
-    const isLatin1 = uint8() === LATIN1;
-    const len = varUInt32();
-    return isLatin1 ? stringLatin1(len) : stringUtf8(len);
+  stringOfVarUInt32() {
+    const isLatin1 = this.uint8() === LATIN1;
+    const len = this.varUInt32();
+    return isLatin1 ? this.stringLatin1(len) : this.stringUtf8(len);
+  }
+
+  stringLatin1(len: number) {
+    if (this.sliceStringEnable) {
+      return this.stringLatin1Fast(len);
+    }
+    return this.stringLatin1Slow(len);
   }
 
-  function stringLatin1Fast(len: number) {
-    const result = bigString.substring(cursor, cursor + len);
-    cursor += len;
+  private stringLatin1Fast(len: number) {
+    const result = this.bigString.substring(this.cursor, this.cursor + len);
+    this.cursor += len;
     return result;
   }
 
-  function stringLatin1Slow(len: number) {
-    const rawCursor = cursor;
-    cursor += len;
-    return readLatin1String(buffer, len, rawCursor);
+  private stringLatin1Slow(len: number) {
+    const rawCursor = this.cursor;
+    this.cursor += len;
+    return readLatin1String(this.platformBuffer, len, rawCursor);
   }
 
-  function binary(len: number) {
+  buffer(len: number) {
     const result = alloc(len);
-    buffer.copy(result, 0, cursor, cursor + len);
-    cursor += len;
+    this.platformBuffer.copy(result, 0, this.cursor, this.cursor + len);
+    this.cursor += len;
     return result;
   }
 
-  function bufferRef(len: number) {
-    const result = buffer.subarray(cursor, cursor + len);
-    cursor += len;
+  bufferRef(len: number) {
+    const result = this.platformBuffer.subarray(this.cursor, this.cursor + len);
+    this.cursor += len;
     return result;
   }
 
-  function varUInt32() {
+  varUInt32() {
     // Reduce memory reads as much as possible. Reading a uint32 at once is far faster than reading four uint8s separately.
-    if (byteLength - cursor >= 5) {
-      const u32 = dataView.getUint32(cursor++, true);
+    if (this.byteLength - this.cursor >= 5) {
```

Comment:
Could we optimize this code using something similar to the [Java implementation](https://github.com/apache/incubator-fury/blob/main/java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java#L1777-L1810)? So that we can reduce some bit move and intermediate variables.

Reply:
Good idea. I will change it.